compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1421,7 +1421,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     plural
                 ),
             );
-            if plural == "" {
+
+            if inexistent_fields.len() == 1 && unmentioned_fields.len() == 1 {
+                let unmentioned_field = unmentioned_fields[0].1.name;
+                err.span_suggestion(
+                    ident.span,
+                    &format!(
+                        "`{}` has a field named `{}`, try replacing `{}` for it.",
+                        tcx.def_path_str(variant.def_id),
+                        unmentioned_field,
+                        inexistent_fields[0]
+                    ),
```

Comment:
What about keeping only the start of the sentence? The error message will already have the name of the inexistant field in it.
```suggestion
                    &format!(
                        "`{}` has a field named `{}`",
                        tcx.def_path_str(variant.def_id),
                        unmentioned_field,
                    ),
```
This will read as "struct XXX does not have a field named YYY. help: XXX has a field named ZZZ", which sounds good to me (I'm definitely not an expert in diagnostics though).

Also, other nit: messages in diagnostics usually don't end with a `.`